sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -351,7 +351,7 @@ abstract class JdbcDialect extends Serializable with Logging {
    */
   @Since("2.3.0")
   protected[jdbc] def escapeSql(value: String): String =
-    if (value == null) null else StringUtils.replace(value, "'", "''")
+    if (value == null) null else StringUtils.replace(value, "'", "\\'")
```

Comment:
To @mihailom-db , I believe this is a breaking change which we should not apply to all JDBCDialect.

If you want to fix something in MySQL, could you isolate it to MySQL only?
> [SPARK-48172][SQL] Fix escaping issue for mysql